Library/Homebrew/cmd/update-report.rb

```diff
@@ -126,132 +127,143 @@ def output_update_report
-      puts "Already up-to-date." unless args.quiet?
-    end
 
-    Commands.rebuild_commands_completion_list
-    link_completions_manpages_and_docs
-    Tap.each(&:link_completions_and_manpages)
+      Commands.rebuild_commands_completion_list
+      link_completions_manpages_and_docs
+      Tap.each(&:link_completions_and_manpages)
 
-    failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
-    if failed_fetch_dirs.present?
-      failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
+      failed_fetch_dirs = ENV["HOMEBREW_MISSING_REMOTE_REF_DIRS"]&.split("\n")
+      if failed_fetch_dirs.present?
+        failed_fetch_taps = failed_fetch_dirs.map { |dir| Tap.from_path(dir) }
 
-      ofail <<~EOS
-        Some taps failed to update!
-        The following taps can not read their remote branches:
-          #{failed_fetch_taps.join("\n  ")}
-        This is happening because the remote branch was renamed or deleted.
-        Reset taps to point to the correct remote branches by running `brew tap --repair`
-      EOS
+        ofail <<~EOS
+          Some taps failed to update!
+          The following taps can not read their remote branches:
+            #{failed_fetch_taps.join("\n  ")}
+          This is happening because the remote branch was renamed or deleted.
+          Reset taps to point to the correct remote branches by running `brew tap --repair`
+        EOS
+      end
     end
 
     return if new_repository_version.blank?
 
     puts
     ohai "Homebrew was updated to version #{new_repository_version}"
-    if new_repository_version.split(".").last == "0"
-      Settings.write "latesttag", new_repository_version
-      puts <<~EOS
-        More detailed release notes are available on the Homebrew Blog:
-          #{Formatter.url("https://brew.sh/blog/#{new_repository_version}")}
-      EOS
-    elsif !args.quiet?
-      Settings.write "latesttag", new_repository_version
+
+    new_major_version, new_minor_version = (new_repository_version.split(".")[0, 2]).map(&:to_i)
+
+    if old_repository_version.present?
+      old_major_version, new_minor_version = (old_repository_version.split(".")[0, 2]).map(&:to_i)
+    end
+
+    if old_major_version.blank? || (new_major_version > old_major_version) || (new_minor_version > old_minor_version)
       puts <<~EOS
-        The changelog can be found at:
-          #{Formatter.url("https://github.com/Homebrew/brew/releases/tag/#{new_repository_version}")}
+        Detailed release notes are available on the Homebrew Blog:
+          #{Formatter.url("https://brew.sh/blog/#{new_major_version}.#{new_minor_version}.0")}
       EOS
     end
+
+    return if new_minor_version.zero? && args.quiet?
+
+    puts <<~EOS
+      The revision changelog can be found at:
+        #{Formatter.url("https://github.com/Homebrew/brew/releases/tag/#{new_repository_version}")}
+    EOS
```

Comment:
I think this should stay as it was previously:

```suggestion
    elsif !args.quiet?
      puts <<~EOS
        The revision changelog can be found at:
          #{Formatter.url("https://github.com/Homebrew/brew/releases/tag/#{new_repository_version}")}
      EOS
    end
```

Reply:
The logic is the same. The unmodified code excluded the revision message if the version was x.x.0.

Reply:
Right, we don't need that. The blog post should be sufficient. If other maintainers disagree I'll yield but I don't think there's a benefit to showing both since it just makes the output longer. I think we even considered not ever showing the GitHub changelog since it's not really useful to most people.